### PR TITLE
PR: add support for equation labelling and referencing (more mathjax compatibility)

### PR DESCRIPTION
**Pull request by qnxor**
As seen here: https://github.com/KaTeX/KaTeX/issues/2003#issuecomment-843991794

I enabled it in the Discourse code too as seen in the commit. This adds support for `\tag, \label, \ref, \eqref` so equations can be labeled and referenced (references are clickable too). This is another feature that Katex disables by default. Enabling it brings the Katex implementation in DIscourse more inline with Mathjax. My previous PR added support for persistent macros.

Tested and working in latest Discourse stable.

@pmusaraj

### CODE REVIEW

**Comment by pmusaraj** [comment]
Thanks @qnxor, can you fix the linting error please?

**Comment by pmusaraj** [comment]
It's the `assets/javascripts/initializers/discourse-math-katex.js` file. You need to: 

- run `yarn install` at the root of the plugin repository
- run `yarn run prettier --write assets/javascripts/initializers/discourse-math-katex.js`

**Comment by qnxor** [comment]
```
~$ cd discourse-math
~/discourse-math$ yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
```
Never bothered with yarn or prettier in the past myself

File: assets/javascripts/initializers/discourse-math-katex.js [hunk 1]
```diff
@@ -38,9 +38,19 @@ function katex($elem) {
...
+      // also enable equation labelling and referencing which are disabled by default
+      // both of these are enabled in mathjax by default, so now the katex implementation is (more) mathjax compatible
+      const katexOpts = {
+        trust: (context) => ['\\htmlId', '\\href'].includes(context.command),
+        macros: {
+          "\\eqref": "\\href{###1}{(\\text{#1})}",
+          "\\ref": "\\href{###1}{\\text{#1}}",
+          "\\label": "\\htmlId{#1}{}"
+        },
+        displayMode: false,
```
Comment (pmusaraj) [inline assets/javascripts/initializers/discourse-math-katex.js:51]: We can probably set the displayMode value based on the element here. This should work (untested)

```suggestion
        displayMode: elem.tagName === "DIV",
```

and we don't need to do the same calculation in the decorator.

**Review by pmusaraj — APPROVED** [review]

**Comment by qnxor** [comment]
```
$ yarn --version
0.32+git
```

About your suggested change, it should work and it's equivalent, but I like to keep initializer code separated from the iterator, otherwise one would argue you should do away with the 'decorate()` function entirely and bring its body into the `katex()` caller. With your suggestion, you now have per-elem processing in both `decorate()` and `katex()`, which I'd vote against.

I'll install prettier manually to output something your linter won't complain about.

**Comment by qnxor** [comment]
Fixed into a new commit. Turns out prettier doesn't like single quoted strings. You can run the checks again and approve it.

**Comment by pmusaraj** [comment]
Thanks @qnxor, merging.
